idea/src/org/jetbrains/kotlin/idea/inspections/collections/UselessCallOnCollectionInspection.kt

```diff
@@ -68,6 +71,11 @@ class UselessCallOnCollectionInspection : AbstractUselessCallInspection() {
             )
             holder.registerProblem(descriptor)
         } else {
+            val fix = if (receiverType.constructor.declarationDescriptor?.fqNameSafe == KotlinBuiltIns.FQ_NAMES.list) {
```

Comment:
You need to check for both conditions. You should replace selector with `toList()` when `whole exp` == `List` && `receiver` != `List`. Now you only check for `receiver` != `List`. So your code doesn't work for `sequenceOf` (because whole expression has type Sequence, not List):
```
val arr: Sequence<Int> = sequenceOf(1, 2, 3).filterNotNull()
```
Yes, we don't yet have "Useless call" check for sequences but your code will broke when we add it.

So I suggest to invert your if for improving readability and add full check.